### ISSUE TITLE
[Feedback]: Strange bool optIn Value Onesignal 5.0.0 stable

### ISSUE THREAD

**Issue by xunreal75** [OneSignal/OneSignal-Flutter-SDK#720]
### What's on your mind?

Hi your optIn bool isn't helpful and hard to understand - please 

I don't understand the sense of the value - it's not logical.

my understand is bool optIn is true if I'm optedOut - why this inverted logic?

```dart

 /// Gets a boolean value indicating whether the current user is opted in to push notifications.
  /// This returns true when the app has notifications permission and optedOut is called.
  /// Note: Does not take into account the existence of the subscription ID and push token.
  /// This boolean may return true but push notifications may still not be received by the user.
  bool? get optedIn {
    return _optedIn;
  }
````

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nan-li** [comment]
Hi @xunreal75,

These scenarios can help understand what the `optedIn` property returns. There is no `optedOut` boolean.
- If the device **does not** have push permission, `optedIn` is false.
- If the device has push permission, but no push token or subscription ID yet, optedIn is `true`.
- If the device has push permission and `optOut()` was **not** called, `optedIn` is true.
- If the device has push permission and `optOut()` **was** called, `optedIn` is false.

**Comment by emawby** [comment]
Fixed in #743

**Comment by nan-li** [comment]
We have updated the documentation, so closing
